src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -183,15 +183,14 @@ public static Date parseDate(String date, boolean isFutureDateAllowed) throws Pa
         requireNonNull(date);
         requireNonNull(isFutureDateAllowed);
         String trimmedDate = date.trim();
-
         //Check if date format is valid.
-        if (!Date.isValidDateFormat(date)) {
+        if (!Date.isValidDateFormat(trimmedDate)) {
             throw new ParseException(Date.MESSAGE_CONSTRAINTS);
         } else if (!Date.isValidDateValue(trimmedDate)) {
             throw new ParseException(String.format(Date.MESSAGE_VALUE_CONSTRAINTS, trimmedDate));
         }
         //Check if date is after current date and if it is allowed.
-        if (Date.isAfterCurrentDate(date) && !isFutureDateAllowed) {
+        if (Date.isAfterCurrentDate(trimmedDate) & !isFutureDateAllowed) {
```

Comment:
Need to change this back to logical AND

```suggestion
        if (Date.isAfterCurrentDate(trimmedDate) && !isFutureDateAllowed) {
```